File: Python/Rayleigh Noise.py
import cv2
import numpy as np
import matplotlib.pyplot as plt

image = cv2.imread('Enter your image path here')
image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
image = image.astype(np.float64) / 255.0
rayleigh_scale = 0.5
rayleigh_noise_image = np.empty_like(image)
for i in range(3):
    rayleigh_noise = np.random.rayleigh(scale=rayleigh_scale, size=image[:, :, i].shape)
    rayleigh_noise_image[:, :, i] = image[:, :, i] + rayleigh_noise
rayleigh_noise_image = np.clip(rayleigh_noise_image, 0, 1)
plt.figure(figsize=(10, 5))
plt.subplot(1, 2, 1)
plt.imshow(image)
plt.title('Original Image')
plt.axis('off')
plt.subplot(1, 2, 2)
plt.imshow(rayleigh_noise_image)
plt.title('Image with Rayleigh Noise')
plt.axis('off')
plt.tight_layout()
plt.show()

File: MATLAB/Rayleigh Noise.m
originalImage = imread('Enter your image path here');
originalImage = im2double(originalImage);

% Add Rayleigh noise
rayleighScale = 0.5;
rayleighNoiseImage = originalImage;

for i = 1:size(originalImage, 3)
    rayleighNoise = raylrnd(rayleighScale, size(originalImage, 1), size(originalImage, 2));
    rayleighNoiseImage(:, :, i) = originalImage(:, :, i) + rayleighNoise;
end

% Clip values to [0, 1] range
rayleighNoiseImage = min(max(rayleighNoiseImage, 0), 1);

% Display original and noisy images
figure;
subplot(1,2,1);
imshow(originalImage);
title('Original Image');

subplot(1,2,2);
imshow(rayleighNoiseImage);
title('Image with Rayleigh Noise');